tests/end-to-end/external-solution-integration/internal/scenario/event_mesh/wrapper_counter_pod.go

```diff
@@ -0,0 +1,22 @@
+package event_mesh
+
+import (
+	"time"
+
+	retrygo "github.com/avast/retry-go"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/testkit"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/testsuite"
+)
+
+const retryAttemptsCount = 240
+const retryDelay = 1 * time.Second
+
+var opts = []retrygo.Option{
+	retrygo.Attempts(retryAttemptsCount),
+	retrygo.Delay(retryDelay),
+}
+
+// NewCheckCounterPod returns a new CheckCounterPod
+func NewWrappedCounterPod(testService *testkit.TestService, expectedValue int) *testsuite.CheckCounterPod {
```

Comment:
(nit) Why don't we move the `opts` to `testsuite.NewCheckCounterPod` ? So we file would not be required at all. Otherwise the only reason for this file is to set a default value which could be set in the original file as well :) You can compare for empty struct and use the defaults then :)

Reply:
The "check_counter_pod" continue to use as default value, like in the past (see in "pkg/retry/retry.go"): 
```
const retryAttemptsCount = 120
const retryDelay = 1 * time.Second
```
But a user of CheckCounterPod  can set new values from outside. I didn't want to change the old behavior of  CheckCounterPod. The Wrapper could set in the future other config values too, overwriting some default values, if necessary.
It's the same "pattern" like in NewWrappedCloudEventClient, which sets its default values too, replacing the default values contained in "pkg/retry/retry.go"

Reply:
Also, maybe have something like inside:

```golang
func GetDefaultRetryOpts() []retrygo.Option {
	return []retrygo.Option{
		retrygo.Attempts(retryAttemptsCount),
		retrygo.Delay(retryDelay),
	}
}
```

and put it inside `kyma/tests/end-to-end/external-solution-integration/pkg/testsuite/check_counter_pod.go`.

Reply:
The caller, in our case NewWrappedCounterPod overwrites the default values of  CheckCounterPod (which gets its own default values from "retry"). This is similar with NewWrappedCloudEventClient implementation. I've used the same "pattern" and the wrapper can always set its config values, overriding the default values of used components if necessary. Of course it can add new values too. I did'n want to change to much now.